src/abstractprobprog.jl

```diff
@@ -8,19 +9,21 @@ Common base type for models expressed as probabilistic programs.
 """
 abstract type AbstractProbabilisticProgram <: AbstractMCMC.AbstractModel end
 
+DensityInterface.hasdensity(::AbstractProbabilisticProgram) = true
+
 
 """
-    logdensity(model, trace)
+    logdensityof(model, trace)
 
 Evaluate the (possibly unnormalized) density of the model specified by the probabilistic program
 in `model`, at specific values for the random variables given through `trace`.
 
 `trace` can be of any supported internal trace type, or a fixed probability expression.
 
-`logdensity` should interact with conditioning and deconditioning in the way required by probability
-theory.
+`logdensityof` should interact with conditioning and deconditioning in the way required by
+probability theory.
 """
-function logdensity end
+DensityInterface.logdensityof
```

Comment:
I think it would be better to add a docstring only to `logdensityof(::AbstractProbabilisticProgram, ::AbstractModelTrace)`.
```suggestion
function DensityInterface.logdensityof(::AbstractProbabilisticProgram, ::AbstractModelTrace) end
```

Reply:
Should we not also deprecate `logdensity` rather than just dropping it completely?

Reply:
I used this form since `function f end` form is the recommended syntax in Julia documentation: https://docs.julialang.org/en/v1/manual/documentation/#Functions-and-Methods-2
This also adds a method to `f`.

It seems one can avoid this (even though it seems the other form is recommended) by specifying
```suggestion
DensityInterface.logdensityof(::AbstractProbabilisticProgram, ::AbstractModelTrace)
```
instead.